includes/gateway/class-omise-payment-billpayment-tesco.php

```diff
@@ -137,84 +137,92 @@ public function display_barcode( $order, $context = 'view' ) {
-					<?php echo $barcode_ref_number; ?>
-				</small>
-
+				<!-- Display for printing only. -->
+				<!-- Hidden by assets/css/omise-billpayment-print.css. -->
 				<?php if ( 'email' !== $context ) : ?>
-					<div class="omise-billpayment-tesco-print-button">
-						<button onClick="window.print()" class="button button-primary">Print barcode</button>
-					</div>
-				<?php endif; ?>
-			</div>
-
-			<!-- Display for printing only. -->
-			<!-- Hidden by assets/css/omise-billpayment-print.css. -->
-			<?php if ( 'email' !== $context ) : ?>
-
-				<div class="omise-billpayment-tesco-print-detail">
-					<div>
-						<h2><?php echo apply_filters( 'woocommerce_thankyou_order_received_text', __( 'Thank you. Your order has been received.', 'woocommerce' ), $order ); ?></h2>
-					</div>
-
 					<div class="omise-billpayment-tesco-print-order-overview">
-						<div class="omise-billpayment-tesco-print-order-overview-item">
+						<div>
 							<?php _e( 'Order number:', 'woocommerce' ); ?>
 							<br/><strong><?php echo $this->order()->get_order_number(); ?></strong>
 						</div>
 
-						<div class="omise-billpayment-tesco-print-order-overview-item">
-							<?php _e( 'Date:', 'woocommerce' ); ?>:
+						<div>
+							<?php _e( 'Date:', 'woocommerce' ); ?>
 							<br/><strong><?php echo wc_format_datetime( $this->order()->get_date_created() ); ?></strong>
 						</div>
 
-						<?php if ( is_user_logged_in() && $this->order()->get_user_id() === get_current_user_id() && $this->order()->get_billing_email() ) : ?>
-							<div class="omise-billpayment-tesco-print-order-overview-item">
-								<?php _e( 'Email:', 'woocommerce' ); ?>:
-								<br/><strong><?php echo $this->order()->get_billing_email(); ?></strong>
-							</div>
-						<?php endif; ?>
-
-						<div class="omise-billpayment-tesco-print-order-overview-item">
-							<?php _e( 'Total:', 'woocommerce' ); ?>:
+						<div>
+							<?php _e( 'Total:', 'woocommerce' ); ?>
 							<br/><strong><?php echo $this->order()->get_formatted_order_total(); ?></strong>
 						</div>
-					</div>
 
-					<?php if ( $this->order()->get_payment_method_title() ) : ?>
-						<div class="omise-billpayment-tesco-print-order-payment-method">
-							<?php _e( 'Payment method:', 'woocommerce' ); ?>
-							<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
-						</div>
-					<?php endif; ?>
+						<?php if ( $this->order()->get_payment_method_title() ) : ?>
+							<div>
+								<?php _e( 'Payment method:', 'woocommerce' ); ?>
+								<br/><strong><?php echo wp_kses_post( $this->order()->get_payment_method_title() ); ?></strong>
+							</div>
+						<?php endif; ?>
+					</div>
+				<?php endif; ?>
+				<!-- /END .omise-billpayment-tesco-print-detail -->
+
+				<p>
+					<?php
+					echo sprintf(
+						wp_kses(
+							__( 'Please bring this barcode to pay at Tesco Lotus within:<br/><strong>%1$s</strong> at <strong>%2$s</strong>.', 'omise' ),
```

Comment:
`'Please bring this barcode to pay at Tesco Lotus by:<br/><strong>%1$s %2$s</strong>.'`

Reply:
Maybe even consider just one formatted datetime string using the stores i18n settings

Reply:
@jonrandy how about `by` > `before`? would it more clear?
`Please bring this barcode to pay at Tesco Lotus before: Aug 15 8.37 am.`

Reply:
'by' is the normal way to say this kind of thing in English

Reply:
👍 ic